.github/workflows/build-and-deploy-react-app.yaml

```diff
@@ -25,17 +25,22 @@ on:
         required: false
         default: ""
         type: string
+      workflow_timeout:
+        description: "Timeout duration in minutes"
+        required: false
+        default: 5
+        type: number
 
 jobs:
 
   build-and-deploy-react-app:
     name: "Prepare, build and deploy react application"
     runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.workflow_timeout }}
     environment: ${{ inputs.environment }}
     permissions:
       id-token: write
       contents: read
-    timeout-minutes: 5
     steps:
       - name: "Checkout code"
         uses: NHSDigital/uec-dos-management/.github/actions/checkout-repository@v0.13
```

Comment:
Looks okay but you will need to rebase from main to pick up the latest tagged version of the code  (@v0.14 I think) and then increment every instance of v0.14 to the next tag eg v0.15... I would create that tag locally and push it BEFORE you push the changes to your branch. That way the workflows will run (cos v0.15 will exist).  Note I would find and replace based on v0.14 not @v0.14 so you pick up the required change to .github/actions/checkout-repository/action.yaml